utils/client.py

```diff
@@ -0,0 +1,183 @@
+                     ('id',), ('id',)],
+    'cause': ['causes', ('', 'id'), ('name',), ('id',), ('id',)],
+    'donor': ['donor', (), (), (), ()],
+    'donors': ['donors', (), (), (), ()],
+    'event': ['events', ('', 'id'), ('name', 'description', 'start_date', 'end_date', 'team_limit',
+                                     'team_building_start', 'team_building_end'), ('id',),
+              ('id',)],
+    'locality': ['localities', ('', 'id'), ('name',), ('id',), ('id',)],
+    'participant': ['participants', ('fbid',), ('fbid',), ('fbid',), ('fbid',)],
+    'record': ['records', ('id',), ('date', 'distance', 'participant_id', 'source_id'), (),
+               ('id',)],
+    'source': ['sources', ('', 'id'), ('name',), ('id',), ('id',)],
+    'sponsor': ['sponsor', (), (), (), ()],
+    'sponsors': ['sponsors', (), (), (), ()],
+    'team': ['teams', ('', 'id'), ('name',), ('id',), ('id',)],
+}
+
+CRUD = {
+    'c': 'POST',
+    'r': 'GET',
+    'u': 'PATCH',
+    'd': 'DELETE',
+}
+
+(API, READ, CREATE, UPDATE, DELETE) = range(5)
+
+
+def construct_url(endpoint):
+    """Build a URL from the scheme, server, and port parameters."""
+    scheme = os.environ.get('SCHEME', 'https')
+    domain = os.environ.get('SERVER', 'akf-causes.subshell.org')
+    port = os.environ.get('PORT', '')
+    if port:
+        netloc = '{}:{}'.format(domain, port)
+    else:
+        netloc = domain
+    url = urlparse.urlunparse((scheme, netloc, endpoint, '', '', ''))
+    return url
+
+
+def get(entity, endpoint, action, url, args):
+    """Construct a GET request for the given entity and parameters."""
+    if not ENTITIES[entity][READ]:
+        print('Method Not Allowed')
+        return None
+
+    if entity == 'achievements':
+        if not args:
+            print('Query achievements with "team_id=<team_id>" \
+                   or "achievement_id=<achievement_id>"')
+            return None
+
+        if args[0].startswith('team'):
+            url += '/team/{}'.format(args[0].split('=')[1])
+        elif args[0].startswith('achievement'):
+            url += '/achievement/{}'.format(args[0].split('=')[1])
+        else:
+            print('Query achievements with "team_id=<team_id>" \
+                   or "achievement_id=<achievement_id>"')
+            return None
+
+        return urllib2.Request(url)
+
+    if args:
+        url += '/{}'.format(args[0])
+    return urllib2.Request(url)
+
+
+def post(entity, endpoint, action, url, args):
+    """Construct a POST request for the given entity and parameters."""
+    if not ENTITIES[entity][CREATE]:
+        print('Method Not Allowed')
+        return None
+
+    if len(args) != len(ENTITIES[entity][CREATE]):
+        print('Create {} requires the following parameters:\n{}'
+              .format(entity, '\n'.join(ENTITIES[entity][CREATE])))
+        return None
+
+    data = {}
+    for key, arg in zip(ENTITIES[entity][CREATE], args):
+        data[key] = arg
```

Comment:
Isnt this equivalent to:

    data = dict(zip(ENTITIES[entity][CREATE], args))